regtest/cluster.js

```diff
@@ -0,0 +1,182 @@
+'use strict';
+
+var path = require('path');
+var async = require('async');
+var spawn = require('child_process').spawn;
+
+var BitcoinRPC = require('bitcoind-rpc');
+var rimraf = require('rimraf');
+var bitcore = require('bitcore-lib');
+var chai = require('chai');
+var should = chai.should();
+
+var index = require('..');
+var log = index.log;
+log.debug = function() {};
+var BitcoreNode = index.Node;
+var BitcoinService = index.services.Bitcoin;
+
+describe('Bitcoin Cluster', function() {
+  var node;
+  var daemons = [];
+  var execPath = path.resolve(__dirname, '../bin/bitcoind');
+  var nodesConf = [
+    {
+      datadir: path.resolve(__dirname, './data/node1'),
+      conf: path.resolve(__dirname, './data/node1/bitcoin.conf'),
+      rpcuser: 'bitcoin',
+      rpcpassword: 'local321',
+      rpcport: 30521,
+      zmqpubrawtx: 'tcp://127.0.0.1:30611',
+      zmqpubhashblock: 'tcp://127.0.0.1:30611'
+    },
+    {
+      datadir: path.resolve(__dirname, './data/node2'),
+      conf: path.resolve(__dirname, './data/node2/bitcoin.conf'),
+      rpcuser: 'bitcoin',
+      rpcpassword: 'local321',
+      rpcport: 30522,
+      zmqpubrawtx: 'tcp://127.0.0.1:30622',
+      zmqpubhashblock: 'tcp://127.0.0.1:30622'
+    },
+    {
+      datadir: path.resolve(__dirname, './data/node3'),
+      conf: path.resolve(__dirname, './data/node3/bitcoin.conf'),
+      rpcuser: 'bitcoin',
+      rpcpassword: 'local321',
+      rpcport: 30523,
+      zmqpubrawtx: 'tcp://127.0.0.1:30633',
+      zmqpubhashblock: 'tcp://127.0.0.1:30633'
+    }
+  ];
+
+  before(function(done) {
+    log.info('Starting 3 bitcoind daemons');
+    this.timeout(20000);
```

Comment:
I tried increasing the timeout to 200000, which is 10 times longer and it still fails, though It does give me a stacktrace.

```bash
$ mocha -R spec regtest/cluster.js


  Bitcoin Cluster
[2016-05-06T19:42:54.013Z] info: Starting 3 bitcoind daemons
    1) "before all" hook
    2) "after all" hook


  0 passing (55s)
  2 failing

  1) Bitcoin Cluster "before all" hook:
     Error: done() invoked with non-Error: {"code":-28,"message":"Loading wallet..."}
      at node_modules/async/lib/async.js:52:16
      at done (node_modules/async/lib/async.js:241:17)
      at node_modules/async/lib/async.js:44:16
      at node_modules/async/lib/async.js:676:51
      at node_modules/async/lib/async.js:726:13
      at node_modules/async/lib/async.js:52:16
      at node_modules/async/lib/async.js:264:21
      at node_modules/async/lib/async.js:44:16
      at node_modules/async/lib/async.js:723:17
      at node_modules/async/lib/async.js:167:37
      at node_modules/async/lib/async.js:652:25
      at IncomingMessage.<anonymous> (node_modules/bitcoind-rpc/lib/index.js:101:7)
      at endReadableNT (_stream_readable.js:913:12)

  2) Bitcoin Cluster "after all" hook:
     Error: timeout of 10000ms exceeded. Ensure the done() callback is being called in this test.
```